### DISCUSSION TITLE
Laravel paginate and inRandomOrder

### DISCUSSION THREAD

**Discussion by omidMolaverdi** [laravel/framework#33821] [General]
- Laravel Version: 6.2.28
- PHP Version:7.4
- Database Driver & Version: Mysql

### Description:
I realised that combining `paginate `and `inRandomOrder `ends up the reverse order in the page numbers. To explain it more, if you have a code like below:

```
       $links = DB::table('linkbox')
            ->inRandomOrder('1234')
            ->paginate(4);
```
and imagine that the table has 7 records. In the view I will see 3 records on page 1 and 4 records on page 2. If I drop `inRandomOrder `then the pagination is right [the right order here should be 4 on page 1 and 3 on page 2].

**Comment by driesvints** [comment]
Moving to discussions.

**Comment by omidMolaverdi** [comment]
hi,


is there any interest to this thread?

**Comment by staudenmeir** [comment]
What queries does this execute for you?

**Reply by omidMolaverdi** [reply]
@staudenmeir Thanks for joining this thread. The query is quite simple and returns **_random_** URLs from a table of URLs. However, as there could be a lot of URLs on the return query, I use pagination. At this point, the issue starts. Assume the return query has a total of 7 URLs and they get split onto 2 pages. The expected result should be 4 results on the first page and 3 results on the second page. However, in this case I see the reverse that is 3 on the first page and 4 on the second page. I copy the code here from the first post above:


```
  $links = DB::table('linkbox')
            ->inRandomOrder('1234')
            ->paginate(4);
```

**Comment by staudenmeir** [comment]
What SQL does this query execute for the first and second page when you log it with `\DB::enableQueryLog(); [...] dump(\DB::getQueryLog());` or a tool like [debugbar](https://github.com/barryvdh/laravel-debugbar)?
